extras/static_code_analysis.sh

```diff
@@ -47,15 +47,15 @@ PERFORM_PEP8=${12}            # true or false, indicating whether PEP8 analysis
 IGNORE_MSG_VERA=${13}         # true or false, indicating whether VERA++ messages should accout for the build result.
 IGNORE_MSG_CPPCHECK=${14}     # true or false, indicating whether CPPCHECK messages should accout for the build result.
 IGNORE_MSG_CLANG_FORMAT=${15} # true or false, indicating whether CLANG-FORMAT messages should accout for the build result.
-IGNORE_MSG_PEP8=${16}         # true or false, indicating whether PEP8 messages should accout for the build result.
+IGNORE_MSG_PYCODESTYLE=${16}  # true or false, indicating whether pycodestyle messages should accout for the build result.
 
-# PEP8 rules to ignore.
-PEP8_IGNORES="E121,E123,E126,E226,E24,E704"
-PEP8_IGNORES_EXAMPLES="${PEP8_IGNORES},E402"
-PEP8_IGNORES_TOPO_MANUAL="${PEP8_IGNORES_EXAMPLES},E265"
+# PYCODESTYLE rules to ignore.
+PYCODESTYLE_IGNORES="E121,E123,E126,E226,E24,E704"
```

Comment:
Please also add W503 and W504 here. See [above](https://github.com/nest/nest-simulator/pull/2105/files#diff-cb3c61b3c773983f77fd537c8923e562835078815e095a308a10d176cfe54a1bR285) for why.

I also dislike it a bit that the verification above uses its own set of ignores and does not use the list defined here. How complicated would it be to do so?

Reply:
You are right. I will make a list of error and warning codes and pass them to this script.